lib/matplotlib/widgets.py

```diff
@@ -1110,24 +1124,36 @@ def onmove(self, event):
             return
         if self.vertOn:
             for line in self.vlines:
-                line.set_xdata((event.xdata, event.xdata))
-                line.set_visible(self.visible)
+                visible = self.visible
+                if not self.vertMulti:
+                    visible = visible and line.axes == event.inaxes
+
+                if visible:
+                    line.set_xdata((event.xdata, event.xdata))
+                    line.set_visible(visible)
         if self.horizOn:
             for line in self.hlines:
-                line.set_ydata((event.ydata, event.ydata))
-                line.set_visible(self.visible)
-        self._update()
+                visible = self.visible
+                if not self.horizMulti:
+                    visible = visible and line.axes == event.inaxes
+                if visible:
+                    line.set_ydata((event.ydata, event.ydata))
+                    line.set_visible(self.visible)
+        self._update(event)
 
-    def _update(self):
+    def _update(self, event):
         if self.useblit:
             if self.background is not None:
                 self.canvas.restore_region(self.background)
             if self.vertOn:
                 for ax, line in zip(self.axes, self.vlines):
-                    ax.draw_artist(line)
+                    if self.vertMulti or event.inaxes == line.axes:
```

Comment:
There is only one background for the whole canvas, we need to redraw all visible lines, not just the ones in the current axes.